client/src/components/RelatedItems/RelatedProducts/RelatedProducts.jsx

```diff
@@ -0,0 +1,248 @@
+import { useState, useEffect } from 'react';
+import { Rate, Modal } from 'antd';
+import { StarOutlined } from '@ant-design/icons';
+import { Virtual, Navigation, Pagination } from 'swiper/modules';
+import { Swiper, SwiperSlide } from 'swiper/react';
+import {
+  getRelatedIds,
+  getDetailById,
+  getReviewById,
+  getProductInfoById,
+} from '@/api/product-api.js';
+import 'swiper/css';
+import 'swiper/css/pagination';
+import 'swiper/css/navigation';
+
+const RelatedProducts = () => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [swiperRef, setSwiperRef] = useState(null);
+  const [slides, setSlides] = useState([]);
+  const [fixedCompareInfo, setFixedCompareInfo] = useState([]);
+  const [features, setFeatures] = useState({});
+  useEffect(() => {
+    getRelatedIdsApi();
+  }, []);
+
+  const getRelatedIdsApi = async () => {
+    try {
+      const res = await getRelatedIds();
+      const finalRes = [];
+
+      res.forEach(async (item, index) => {
```

Comment:
Is the response returned by getRelatedIds() an iterable item?
I think you want to do res.data.forEach.